script.js

```diff
@@ -0,0 +1,364 @@
+initializeCalculator();
+function initializeCalculator() {
+  let first = {value: "", unsaved: true, saved: false, integer: true};
+  let second = {value: "", unsaved: true, saved: false, integer: true};
+  let operator = {value: "", unsaved: true, saved: false};
+  let result = "";
+  let allowNewNumber = false;
+
+  listenClicks();
+  listenKeys();
+
+  function listenKeys() {
+    window.addEventListener("keydown", event => {
+      const key = document.querySelector(`[data-key="${event.key}"]`);
+      if (key) {
+        bindEvent(key.id);
+        key.focus(); // Safari and Firefox don't handle focus in CSS files
+      }
+    });
+  }
+
+  function listenClicks() {
+    const buttons = document.querySelectorAll(".btn");
+    buttons.forEach(button => {
+      button.addEventListener("click", event => {
+        const click = event.target;
+        bindEvent(click.id);
+        click.focus(); // Safari and Firefox don't handle focus in CSS files
+      });
+    });
+  }
+
+  function bindEvent(event) {
```

Comment:
What you're calling `event` here is not an event but `keyValue`, could we update the name to be more precise?